src/openassetio-core/include/openassetio/Context.hpp

```diff
@@ -41,10 +41,32 @@ class OPENASSETIO_CORE_EXPORT Context final {
   /**
    * @name Access Pattern
    */
-  enum class Access { kRead, kReadMultiple, kWrite, kWriteMultiple, kUnknown };
+  enum class Access {
+    /**
+     * Host intends to read data
+     */
+    kRead,
+    /**
+     * Host intends to write data. This should be the default choice for
+     * write, unless the conditions for @ref Access.kCreateRelated are
+     * met.
```

Comment:
[fix] Attempting to capture the "what is the reference for" aspect:
```suggestion
     * Host intends to write data.  For example, trait property values obtained 
       via `resolve` may be used to control the writing of data to a resource, 
       and specifics of its format or content.
       During publishing this must be used whenever the entity reference 
       explicity targets the specific entity whose data is being written. 
       For example creating or updating a simple, unstructured asset such
       as an image or other file-based data.
```

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1036/commits/dbf716a78430f6d2f622684718ee844dfdc0435d

Gone with this cause you're the source of truth here, but these docs, (the above one for kRead as well,) are more confusing than illuminating to me personally.

Specifically,
> For example, trait property values obtained via `resolve` may be used to control the writing of data to a resource.

I don't see how this is an example of a host intending to write data, nor do I understand what a trait-property-value being used as a control means.

>  During publishing this must be used whenever the entity reference explicity targets the specific entity whose data is being written. For example creating or updating a simple, unstructured asset such as an image or other file-based data

The first part of that reads _okay_ to me, although it does seem to imply that there's an alternate you would consider as an equal partner to `kWrite`, but the second part, "creating or updating a simple, unstructured asset", I feel strongly implies for more complicated cases there is an alternate you should be looking at, but as far as I know, `kWrite` is the correct applicable context for the vast majority of publish workflows.

As ever, this is likely my fault for being slow, and we should just go with what you've written to get this through.

Reply:
> I don't see how this is an example of a host intending to write data, nor did I really understand what a trait-property-value being used as a control means without thinking about it for a hot minute. I'm guessing you're referring to the idea that a trait value might allow the manager to branch their storage mechanism or reaction based on the value stored? 

e.g The Host uses the resolved value of the LocatableContent trait's location property to determine where to write the file to. Many pipelines put working data in a different place to its final resting place. 

> I feel strongly implies for more complicated cases there is an alternate you should be looking at, but as far as I know, kWrite is the correct applicable context for the vast majority of publish workflows.

kCreateRelated is what you use when you are creating components of a structured asset based on some anchor. So for anything where a host is making more than one thing for against a single reference.

Not being slow at all! Holidays etc have not been kind to us of late in terms of overlap!